### PR TITLE
add logic to determine webproj, fix issue #2344

### PR DESCRIPTION
**Pull request by watashiSHUN**
since there's fair amount of reorganize, I've ran all private tests to make sure it does not break anything. I also tested against WebAPI project without `wwwroot/web.config`

TODO: add repo to [kudu test repos](https://github.com/KuduApps)

### CODE REVIEW

File: Kudu.Core/Infrastructure/AspNetCoreHelper.cs [hunk 1]
```diff
@@ -40,10 +41,22 @@ public static bool TryAspNetCoreWebProject(string rootPath, IFileFinder fileFind
...
         {
-            // we need to verify suffix is csproj, xproj will not have projectTypeGuids either
-            return projectPath.EndsWith(".csproj", StringComparison.OrdinalIgnoreCase) && !projectTypeGuids.Any() && IsWebApplicationProjectFile(projectPath);
+            if (projectPath.EndsWith(".csproj", StringComparison.OrdinalIgnoreCase))
+            {
+                // for dotnet core preview 3 and after
+                // if its .csproj, look for <PackageReference Include="Microsoft.AspNetCore" Version="..." />
+                return !projectTypeGuids.Any() &&
+                (VsHelper.DoesIncludeReferencePackage(projectPath, "Microsoft.AspNetCore")
+                || IsWebAppFromFolderStruct(projectPath));
```
Comment (davidebbo) [inline Kudu.Core/Infrastructure/AspNetCoreHelper.cs]: nit: Indent the 2 lines to show it's the same statement.

File: Kudu.Core/Infrastructure/AspNetCoreHelper.cs [hunk 2]
```diff
@@ -40,10 +41,22 @@ public static bool TryAspNetCoreWebProject(string rootPath, IFileFinder fileFind
...
+            }
+            else if (projectPath.EndsWith(".xproj", StringComparison.OrdinalIgnoreCase))
+            {
+                // for dotnet core preview 2 and before 
+                return IsWebAppFromFolderStruct(projectPath);
```
Comment (davidebbo) [inline Kudu.Core/Infrastructure/AspNetCoreHelper.cs]: So `projectTypeGuids` don't apply in this case?

File: Kudu.Core/Infrastructure/VsHelper.cs [hunk 3]
```diff
@@ -82,6 +82,17 @@ public static IEnumerable<Guid> GetProjectTypeGuids(string path)
                         select new Guid(guid.Trim('{', '}'));
             return guids;
         }
+
+        public static bool DoesIncludeReferencePackage(string path, string packageName)
```
Comment (davidebbo) [inline Kudu.Core/Infrastructure/VsHelper.cs]: nit: more typical name is simply `IncludesReferencePackage`

File: Kudu.Core/Infrastructure/AspNetCoreHelper.cs (hunk 2 above)
Comment (watashiSHUN) [inline Kudu.Core/Infrastructure/AspNetCoreHelper.cs]: if it ends with xproj, we know its dotnet core preview2 or before

**Pull request closed without merging**